### PR TITLE
Revert "Revert "List items rendered""

### PR DESCRIPTION
**Pull request by faysvas**
Reverts the-collab-lab/tcl-49-smart-shopping-list#18 which reverted the-collab-lab/tcl-49-smart-shopping-list/pull/16. This reopens the original PR

### CODE REVIEW

File: src/views/List.jsx [hunk 1]
```diff
@@ -7,6 +8,10 @@ export function List({ data }) {
 				Hello from the <code>/list</code> page!
 			</p>
 			<ul>
+				{React.Children.toArray(
```
Comment (pmbanugo) [inline src/views/List.jsx]: Why did you choose `React.Children.toArray` over rendering a list of JSX element using `data.map`? 

You could map over the array items and return JSX like you already have inside React.Children.toArray() and it should also work.

**Review by pmbanugo — COMMENTED** [review]
Good job 👍🏽  Just a minor comment to clarify.

File: src/views/List.jsx [hunk 2]
```diff
@@ -7,6 +8,10 @@ export function List({ data }) {
...
+					data.map(({ name }) => <ListItem name={name} />),
```
Comment (faysvas) [inline src/views/List.jsx]: Also, it is a good practice to define a [key](https://reactjs.org/docs/lists-and-keys.html#keys) for each item when you have a list of items :slightly_smiling_face:

File: src/views/List.jsx [hunk 3]
```diff
@@ -7,6 +8,10 @@ export function List({ data }) {
...
+				)}
+
 				{/**
 				 * TODO: write some JavaScript that renders the `data` array
 				 * using the `ListItem` component that's imported at the top
```
Comment (faysvas) [inline src/views/List.jsx]: Please delete this TODO comment since you don't need it anymore :slightly_smiling_face:

Comment (Terieyenike) [reply src/views/List.jsx]: We will do just that. 
Thank you for highlighting it.

File: src/views/List.jsx (hunk 2 above)
Comment (Terieyenike) [reply src/views/List.jsx]: We chose the `React.Children.toArray` method to automatically allow React to assign the keys when rendering a list of children. But it is OK, as @Mama-simba and I can reverse it to the method we all know. 

@pmbanugo please check this.

**Review by pmbanugo — APPROVED** [review]
Well done @Terieyenike @Mama-simba 👏🏽

File: src/views/List.jsx (hunk 2 above)
Comment (faysvas) [reply src/views/List.jsx]: Oh, cool, I didn't know that :smiley:

**Pull request merged**